src/main/java/cloud/eppo/ConfigurationRequestor.java

```diff
@@ -1,55 +1,148 @@
 package cloud.eppo;
 
-import java.io.IOException;
-import okhttp3.Response;
+import cloud.eppo.api.Configuration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+import org.jetbrains.annotations.NotNull;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-// TODO: handle bandit stuff
 public class ConfigurationRequestor {
   private static final Logger log = LoggerFactory.getLogger(ConfigurationRequestor.class);
+  private static final String FLAG_CONFIG_PATH = "/api/flag-config/v1/config";
+  private static final String BANDIT_PARAMETER_PATH = "/api/flag-config/v1/bandits";
 
   private final EppoHttpClient client;
-  private final ConfigurationStore configurationStore;
+  private final IConfigurationStore configurationStore;
   private final boolean expectObfuscatedConfig;
+  private final boolean supportBandits;
+
+  private CompletableFuture<Void> remoteFetchFuture = null;
+  private CompletableFuture<Void> configurationFuture = null;
+  private boolean initialConfigSet = false;
+
+  public ConfigurationRequestor(
+      @NotNull IConfigurationStore configurationStore,
+      @NotNull EppoHttpClient client,
+      boolean expectObfuscatedConfig,
+      boolean supportBandits) {
+    this.configurationStore = configurationStore;
+    this.client = client;
+    this.expectObfuscatedConfig = expectObfuscatedConfig;
+    this.supportBandits = supportBandits;
+  }
 
   public ConfigurationRequestor(
-      ConfigurationStore configurationStore,
-      EppoHttpClient client,
+      @NotNull IConfigurationStore configurationStore,
+      @NotNull EppoHttpClient client,
       boolean expectObfuscatedConfig) {
     this.configurationStore = configurationStore;
     this.client = client;
     this.expectObfuscatedConfig = expectObfuscatedConfig;
+    this.supportBandits = true;
```

Comment:
Same here, I think we can just deal with a singular constructor. The "developer experience" win is limited to our SDK code and doesn't seem worth the added overhead of multiple constructors.